modules/packages/MPI.chpl

```diff
@@ -0,0 +1,370 @@
+/* MPI Bindings for Chapel.
+
+These implement the C-API for the MPI 1.1 standard. We currently do not support the following 
+routines ::
+
+      MPI_Op_create
+      MPI_Op_free
+      MPI_Errhandler_create
+      MPI_Errhandler_free
+      MPI_Keyval_create
+      MPI_Keyval_free
+
+since all of these are built around user-defined handlers, that we support.
+
+.. note::
+  #. Pointer arguments are written as `ref` areguments, so no casting to a C
+     style pointer is necessary.
+  #. An exception to the above is if the C prototype names the argument ``array_of_*``,
+     in which case we write it using an array form.
+  #. Some MPI-1.1 functions were deprecated in MPI-2. These should be updated in the future, but
+     are still present in this version.
+  #. We represent MPI_Aint by ptrdiff. If this is not the correct size, there will be an assertion 
+     failure in the code.
+*/
+module MPI {
+  use SysCTypes;
+  require "mpi.h";
+
+  /* Automatically initialize, and define
+   worldSize and worldRank, since we will
+   likely use those often. 
+
+   You are still responsible for calling MPI_Finalize,
+   until Chapel has a module termination scheme.
```

Comment:
As an aside, I think the Chapel compiler has logic to call record destructors for global variables of record type. I believe this is similar to what you'd do in C++. You could implement calling MPI_Finalize and MPI_Init with such a record. I don't think it's critical for this PR though.